### PR TITLE
Add missing example for Thread struct

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

**Comment by frewsxcv** [comment]
Travis found some issues.

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -610,6 +625,21 @@ struct Inner {
...
+///
+/// let handler = thread::Builder::new()
+///     .spawn(|| {
+///         let thread = thread::current();
+///         println!("thread id: {}", thread.id());
+///     })
+///     .unwrap();
+///
+/// handler.join().unwrap();
+/// ```
```
Comment (frewsxcv) [inline src/libstd/thread/mod.rs]: Where is `Thread` used in this example?

File: src/libstd/thread/mod.rs [hunk 2]
```diff
@@ -633,6 +663,21 @@ impl Thread {
...
+    ///
+    /// let handler = thread::Builder::new()
+    ///     .spawn(|| {
+    ///         let thread = thread::current();
+    ///         thread.unpark();
+    ///     })
+    ///     .unwrap();
+    ///
+    /// handler.join().unwrap();
+    /// ```
```
Comment (frewsxcv) [inline src/libstd/thread/mod.rs]: What is `unpark` doing in this example?

File: src/libstd/thread/mod.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libstd/thread/mod.rs]: In the `spawn` closure.

File: src/libstd/thread/mod.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/libstd/thread/mod.rs]: Not much, I really wonder how to show such an example of usage actually... :-/

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/libstd/thread/mod.rs [hunk 3]
```diff
@@ -610,6 +627,22 @@ struct Inner {
...
 #[stable(feature = "rust1", since = "1.0.0")]
 /// A handle to a thread.
+///
+/// # Examples
+///
+/// ```
+/// use std::thread;
+///
+/// let handler = thread::Builder::new()
+///     .name("foo".into());
```
Comment (frewsxcv) [inline src/libstd/thread/mod.rs]: Semicolon

File: src/libstd/thread/mod.rs [hunk 4]
```diff
@@ -643,6 +691,23 @@ impl Thread {
...
+    ///
+    /// ```
+    /// #![feature(thread_id)]
+    ///
+    /// use std::thread;
+    ///
+    /// let handler = thread::Builder::new()
+    ///     .spawn(|| {
+    ///         let thread = thread::current();
+    ///         println!("thread id: {}", thread.id());
```
Comment (frewsxcv) [inline src/libstd/thread/mod.rs]: `ThreadId` does not implement `Display`

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**